packages/astro/src/core/config.ts

```diff
@@ -123,11 +122,18 @@ interface LoadConfigOptions {
 /** Attempt to load an `astro.config.mjs` file */
 export async function loadConfig(options: LoadConfigOptions): Promise<AstroConfig> {
   const root = options.cwd ? path.resolve(options.cwd) : process.cwd();
-  const astroConfigPath = new URL(`./${options.filename || 'astro.config.mjs'}`, `file://${root}/`);
   let userConfig: AstroUserConfig = {};
-  // Load a user-config, if one exists and is provided
-  if (existsSync(astroConfigPath)) {
-    userConfig = (await import(astroConfigPath.href)).default;
+  let userConfigPath: string|undefined;
+
+  if (options.filename) {
+    userConfigPath = options.filename.startsWith('.') ? options.filename : `./${options.filename}`;
```

Comment:
Should this also check for absolute paths?

```suggestion
    userConfigPath = /^\.*\//.test(options.filename) ? options.filename : `./${options.filename}`;
```